### PR TITLE
fix review

### PR DESCRIPTION
**Pull request by ludete**

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -53,7 +54,7 @@ contract Vault is IVault, Initializable {
   }
 
   /// @inheritdoc IVault
-  function updatePeriod() external returns (uint256 _period) {
+  function emissionReward() external returns (uint256 _period) {
```
Comment (CoderZhi) [inline contracts/Vault.sol:57]: emission is a noun, emit is its verb form

File: contracts/Vault.sol [hunk 2]
```diff
@@ -67,9 +68,10 @@ contract Vault is IVault, Initializable {
       if (_balanceOf < _emission) {
         revert InsufficientFund();
       }
-      uint256 _veEmission = (_emission * veRate) / 10000;
-
-      rewardsDistributor.distributeRewards{value: _veEmission}();
+      uint256 _veEmission = (_emission * shareRate) / 10000;
+      if (_veEmission == 0) revert ZeroEmission();
+      IStrategyManager(strategyManager).distributeRewards{value: _veEmission}(IOTX_NATIVE_TOKEN, _veEmission);
+      emit Emission(msg.sender, _emission);
```
Comment (CoderZhi) [inline contracts/Vault.sol:74]: delete

File: contracts/Vault.sol [hunk 3]
```diff
@@ -67,9 +68,10 @@ contract Vault is IVault, Initializable {
       if (_balanceOf < _emission) {
         revert InsufficientFund();
       }
-      uint256 _veEmission = (_emission * veRate) / 10000;
-
-      rewardsDistributor.distributeRewards{value: _veEmission}();
+      uint256 _veEmission = (_emission * shareRate) / 10000;
+      if (_veEmission == 0) revert ZeroEmission();
```
Comment (CoderZhi) [inline contracts/Vault.sol:72]: if emission is 0, skip reward distribution call.

File: contracts/Vault.sol (hunk 1 above)
Comment (ludete) [reply contracts/Vault.sol:57]: ok

File: contracts/Vault.sol (hunk 2 above)
Comment (ludete) [reply contracts/Vault.sol:74]: ok

File: contracts/Vault.sol (hunk 3 above)
Comment (ludete) [reply contracts/Vault.sol]: ok

**Pull request merged**